Mock Security Realm, RBAC, and Plugins

schottsfired opened the pull request:
Pull request also includes docker-compose.yml with my values, so you'll need to update again. No apparent way to exclude. Maybe we can pull the docker-compose.yml local directories from an environment variable or something.

Users (pw is same as username):
admin - full access
dev - developer access (see roles > manage as admin)
specialdev - developer access plus able to see 'secret-project' folder on apiteam master. see 'secret-project' > groups
read - read-only (authenticated) access

Also includes all plugin .jpi files. It's a bit heavyweight but I've confirmed that it works for versioning new downloaded plugins.

schottsfired committed:
Updated for local setup

schottsfired committed:
more ignores

schottsfired committed:
more ignores

schottsfired committed:
Merge branch 'master' into auth

schottsfired committed:
set up mock security realm and RBAC

schottsfired committed:
User Setup

Created users: admin, dev, specialdev, read
Better .gitignore
Check in plugin binaries

schottsfired committed:
Merge branch 'auth'

schottsfired committed:
Local Slave

Created local slave on apiteam master, see Manage Jenkins > Manage Nodes

This local slave can be used to demo folders plus:
http://jenkins-enterprise.cloudbees.com/docs/user-guide-docs/folders-plu
s.html

kmadel commented on docker-compose.yml:
```diff
@@ -4,7 +4,7 @@ dnsdock:
     volumes: 
         - /var/run/docker.sock:/var/run/docker.sock
-        - /Users/kmadel/dev/CloudBees/docker/cbdemo:/data
+        - /Users/David/Projects/CloudBees/docker/cbdemo:/data
```
We need to figure out a strategy around user specific paths - could see if ~ works, and then align on a standard directory structure: `~/projects/cloudbees/docker/cbdemo`

schottsfired replied:
Agree. Can you test it :)

kmadel commented on docker-compose.yml:
```diff
@@ -70,7 +77,7 @@ jocproxy:
         - joc2
     volumes:
-        - /Users/kmadel/dev/CloudBees/docker/cbdemo/haproxy/joc:/haproxy-override
+        - /Users/David/Projects/CloudBees/docker/cbdemo/haproxy/joc:/haproxy-override
```
see above regarding user specific paths

schottsfired committed:
Update Center

Created custom update center for apiteam (in JOC) and promoted the
‘beer’ plugin :)
